src/main/java/springboot/repository/impl/BookRepositoryImpl.java

```diff
@@ -1,52 +1,59 @@
 package springboot.repository.impl;
 
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
 import java.util.List;
+import java.util.Optional;
 import lombok.RequiredArgsConstructor;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
-import org.hibernate.query.Query;
 import org.springframework.stereotype.Repository;
+import springboot.dto.BookDto;
+import springboot.dto.CreateBookRequestDto;
 import springboot.exeptions.DataProcessingException;
+import springboot.mapper.BookMapper;
 import springboot.model.Book;
 import springboot.repository.BookRepository;
 
 @Repository
 @RequiredArgsConstructor
 public class BookRepositoryImpl implements BookRepository {
 
-    private final SessionFactory sessionFactory;
+    private final EntityManagerFactory entityManagerFactory;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        Session session = null;
-        Transaction transaction = null;
-        try {
-            session = sessionFactory.openSession();
-            transaction = session.beginTransaction();
-            session.persist(book);
+    public BookDto createBook(CreateBookRequestDto requestDto) {
```

Comment:
revert changes

Reply:
Return the standard save method, and in other classes, use createBook, inside which save from BookRepository will be called?